Samples.Objc/Samples.ObjC/ViewControllers/AddItemViewController/Views/QuantityView.m

```diff
@@ -0,0 +1,106 @@
+  [self setupConstraints];
+  return self;
+}
+
+- (void)setupUI {
+  self.minusButton = [[UIButton alloc] init];
+  self.minusButton.tintColor = [UIColor whiteColor];
+  [self.minusButton setTitle:@"-" forState:UIControlStateNormal];
+  [self.minusButton.titleLabel setFont:[UIFont boldSystemFontOfSize:16]];
+  [self.minusButton addTarget:self action:@selector(tapMinus) forControlEvents:UIControlEventTouchUpInside];
+  self.minusButton.translatesAutoresizingMaskIntoConstraints = false;
+  self.minusButton.layer.cornerRadius = 8;
+  [self setupMinusButtonColor];
+  
+  self.plusButton = [[UIButton alloc] init];
+  self.plusButton.backgroundColor = [UIColor systemBlueColor];
+  self.plusButton.tintColor = [UIColor whiteColor];
+  [self.plusButton setTitle:@"+" forState:UIControlStateNormal];
+  [self.plusButton.titleLabel setFont:[UIFont boldSystemFontOfSize:16]];
+  [self.plusButton addTarget:self action:@selector(tapPlus) forControlEvents:UIControlEventTouchUpInside];
+  self.plusButton.translatesAutoresizingMaskIntoConstraints = false;
+  self.plusButton.layer.cornerRadius = 8;
+  
+  self.quantityLabel = [[UILabel alloc] init];
+  self.quantityLabel.font = [UIFont systemFontOfSize:16];
+  self.quantityLabel.textAlignment = NSTextAlignmentCenter;
+  [self.quantityLabel sizeToFit];
+  [self setQuantityText];
+  self.quantityLabel.translatesAutoresizingMaskIntoConstraints = false;
+  
+  self.stackView = [[UIStackView alloc] init];
+  self.stackView.axis = UILayoutConstraintAxisHorizontal;
+  self.stackView.distribution = UIStackViewDistributionEqualSpacing;
+  self.stackView.alignment = UIStackViewAlignmentFill;
+  self.stackView.spacing = 0;
+  [self.stackView addArrangedSubview:self.minusButton];
+  [self.stackView addArrangedSubview:self.quantityLabel];
+  [self.stackView addArrangedSubview:self.plusButton];
+
+  self.stackView.translatesAutoresizingMaskIntoConstraints = false;
+
+  [self addSubview:self.stackView];
+}
+
+- (void)setupConstraints {
+  [[self.minusButton.heightAnchor constraintEqualToAnchor:self.minusButton.widthAnchor] setActive:true];
+  
+  [[self.plusButton.heightAnchor constraintEqualToAnchor:self.plusButton.widthAnchor] setActive:true];
+  
+  [[self.stackView.topAnchor constraintEqualToAnchor:self.topAnchor] setActive:true];
+  [[self.stackView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor] setActive:true];
+  [[self.stackView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor] setActive:true];
+  [[self.stackView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor] setActive:true];
+}
+
+- (void)setupMinusButtonColor {
+  if (self.quantity > 1) {
+    self.minusButton.backgroundColor = [UIColor systemBlueColor];
+  } else {
+    self.minusButton.backgroundColor = [[UIColor systemBlueColor] colorWithAlphaComponent:0.5];
+  }
+}
+
+- (void)tapMinus {
+  if (self.quantity > 1) {
+    self.quantity -= 1;
+    [self setQuantityText];
+    [self setupMinusButtonColor];
+  }
+}
+
+- (void)tapPlus {
```

Comment:
We can be a bit clearer about what is happening here i think - this function is triggered in response to the user tapping the button. `tapPlus` indicates that the function is what does the tapping (mostly because prefixing function names with a verb makes makes me think that the verb is what happens inside the function)

long way of saying that this should instead be named something like `didTapPlus`, `addButtonTapped`, or something like that

Reply:
ohhhh so nice!!!!!! yea that makes sense!!!!